src/spec_tools/json_schema.cljc

```diff
@@ -202,7 +202,10 @@
 (defmethod accept-spec 'clojure.spec/and [dispatch spec children]
   (simplify-all-of {:allOf children}))
 
-; merge
+(defmethod accept-spec 'clojure.spec/merge [dispatch spec children]
+  {:type "object"
+   :properties (apply merge (map :properties children))
+   :required (apply concat (map :required children))})
```

Comment:
Required is a vector elsewhere, so should be here also. Concat returns a lazy seq.

e.g. `(reduce into [] (map :required children))` should quite efficiently add all the children to a single vector.

Also, I think there should be a test to check what happens when several children define the same key.